apps/meteor/server/methods/readThreads.js

```diff
@@ -28,6 +29,7 @@ Meteor.methods({
 			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'getThreadMessages' });
 		}
 
-		return readThread({ userId: user._id, rid: thread.rid, tmid });
+		readThread({ userId: user._id, rid: thread.rid, tmid });
+		Promise.await(markRoomAsRead(thread.rid, user._id));
```

Comment:
marking the room as read might create undesired side effects, like updating the sidebar item..

I'd recommend to call `ReadReceipt.markMessagesAsRead` directly, but this will also generate weird behaviors, like adding read receipts to threads that are older but not read..